src/components/RegisterForm.jsx

```diff
@@ -1,4 +1,160 @@
+    } = this.state;
+    const { password1, password2 } = this.state;
+    email = email.trim();
+    username = username.trim();
+    const errors = validate(username, email, password1, password2);
+    if (Object.keys(errors).length === 0) {
+      store.dispatch({ type: IS_LOADING, payload: { isLoading: true } });
+      const payload = {
+        user: {
+          email,
+          username,
+          password: password2,
+          callback_url: REACT_APP_FRONTEND_URL,
+        },
+      };
+
+      const { RegisterUser } = this.props;
+      RegisterUser(payload);
+    } else {
+      store.dispatch(showErrors(errors));
+    }
+  }
+
+  render() {
+    const { errors, isLoading } = this.props;
+    return (
+      <div className="center">
+        <Form onSubmit={this.handleSubmit}>
+          <Form.Group>
+            <Form.Label>Email</Form.Label>
+            <Form.Text id="email-error" className="error-text">{errors.email}</Form.Text>
+            <Form.Control
+              className={errors.email ? 'error' : ''}
+              type="text"
+              id="email"
+              onChange={this.handleChange}
+              onFocus={this.handleFocus}
+            />
+          </Form.Group>
+          <Form.Group>
+            <Form.Label>Username</Form.Label>
+            <Form.Text id="username-error" className="error-text">{errors.username}</Form.Text>
+            <Form.Control
+              className={errors.username ? 'error' : ''}
+              type="text"
+              id="username"
+              onChange={this.handleChange}
+              onFocus={this.handleFocus}
+            />
+          </Form.Group>
+          <Form.Group>
+            <Form.Label>Password</Form.Label>
+            <Form.Text id="password1-error" className="error-text">{errors.password1}</Form.Text>
+            <Form.Control
+              className={errors.password1 ? 'error' : ''}
+              type="password"
+              id="password1"
+              onChange={this.handleChange}
+              onFocus={this.handleFocus}
+            />
+          </Form.Group>
+          <Form.Group>
+            <Form.Label>Confirm Password</Form.Label>
+            <Form.Text id="password2-error" className="error-text">{errors.password2}</Form.Text>
+            <Form.Control
+              className={errors.password2 ? 'error' : ''}
+              type="password"
+              id="password2"
+              onChange={this.handleChange}
+              onFocus={this.handleFocus}
+            />
+          </Form.Group>
+          { isLoading ? (
+            <Button className="btn-disabled btn-one">
+              <Spinner
+                as="span"
+                animation="grow"
+                size="sm"
+                role="status"
+                aria-hidden="true"
+                className="btn-disabled"
+              />
+           Loading...
+            </Button>
+          ) : <Button className="btn-one" type="submit">REGISTER</Button>}
```

Comment:
@ElMonstro, use one button here disabled can be true or false. Make the <Spinner/> component in a different file and render it in the button only when false. You can use `disabled = isLoading` and `isLoading && <Spinner/>` in the button. This reduces the amount of code, separates the file and makes the code more maintainable.